### PR TITLE
replace UserChecker checkPostAuth by checkPreAuth in LoginManager

### PR DESCRIPTION
**Pull request by whitewhidow**
same as : https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1673   but mergeable to master without conflicts

### CODE REVIEW

File: EventListener/AuthenticationListener.php [hunk 1]
```diff
@@ -42,9 +42,9 @@ public static function getSubscribedEvents()
     public function authenticate(FilterUserResponseEvent $event, $eventName = null, EventDispatcherInterface $eventDispatcher = null)
     {
         if (!$event->getUser()->isEnabled()) {
```
Comment (sstok) [inline EventListener/AuthenticationListener.php]: You should remove this if-statement if not used anymore.

Comment (whitewhidow) [inline EventListener/AuthenticationListener.php]: haha,   yes,   but that makes the tests fail,  so should i change the test ?

Comment (sstok) [inline EventListener/AuthenticationListener.php]: Yes they should be updated.

**Comment by whitewhidow** [comment]
there,  i think all is well now.

**Comment by whitewhidow** [comment]
go baby go

**Comment by sstok** [comment]
Seems good to me.

@stof ping

**Comment by hellomedia** [comment]
thanks @whitewhidow for helping this PR move faster ! I wonder about the tests tho. The account flags check is now handled in `checkPreAuth` , shouldn't there be a test checking for that ? It seems like an important thing to test for. trying to address that in the original PR : https://github.com/FriendsOfSymfony/FOSUserBundle/pull/1673

**Comment by whitewhidow** [comment]
@stof is this ready to be merged,   or do we need an additional test as attempted in PR : #1673

**Comment by whitewhidow** [comment]
any update on this anyone ?

**Comment by whitewhidow** [comment]
bump..

**Comment by archie18** [comment]
Come on guys, why has this not yet been merged?
You have got a security hole as wide as a football goal and do not care to close it?

**Comment by stof** [comment]
I think both checkPreAuth and checkPostAuth should be checked actually.

**Comment by hellomedia** [comment]
in [checkPostAuth](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Core/User/UserChecker.php) , it is checking if the credentials are expired ( must change password after a certain time). Should this prevent login ? If so, what is the conceptual difference between checkPreAuth and checkPostAuth ? Par of the answer might be [here](https://github.com/symfony/symfony/pull/9902) . Personnally, I don't know but I am not sure checkPostAuth should block this PR.

**Comment by stof** [comment]
@hellomedia in Symfony, both are preventing the login

**Comment by hellomedia** [comment]
OK well then, let's add checkPostauth too. I'll fix #1673 this weekend if this one doesn't move.

**Comment by romaricdrigon** [comment]
Any reasons it is not getting merged?
Related to #1983

**Comment by whitewhidow** [comment]
bump ?

**Comment by archie18** [comment]
bump?

**Comment by whitewhidow** [comment]
badabump

**Comment by kolinko-ant** [comment]
Any updates? second PR is stucked

**Comment by XWB** [comment]
I squashed all commits into a single PR: https://github.com/FriendsOfSymfony/FOSUserBundle/pull/2210

**Pull request closed without merging**